src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -100,16 +106,30 @@ private function getIsGrantedSubject(string|Expression $subjectRef, Request $req
 
     private function getIsGrantedString(IsGranted $isGranted): string
     {
-        $processValue = fn ($value) => \sprintf($value instanceof Expression ? 'new Expression("%s")' : '"%s"', $value);
+        $processValue = function ($value) {
+            if ($value instanceof Expression) {
+                return \sprintf('new Expression("%s")', $value);
+            }
+
+            if ($value instanceof \Closure) {
+                return 'callable';
+            }
+
+            return \sprintf('"%s"', $value);
+        };
 
         $argsString = $processValue($isGranted->attribute);
 
         if (null !== $subject = $isGranted->subject) {
-            $subject = !\is_array($subject) ? $processValue($subject) : array_map(function ($key, $value) use ($processValue) {
-                $value = $processValue($value);
-
-                return \is_string($key) ? \sprintf('"%s" => %s', $key, $value) : $value;
-            }, array_keys($subject), $subject);
+            if ($subject instanceof \Closure) {
```

Comment:
This special case is not needed. The existing logic using `$processValue` will handle that case properly already.